test.js

```diff
@@ -47,6 +47,18 @@ test('.set() - with object', t => {
 	t.is(t.context.config.get('baz.foo.bar'), 'baz');
 });
 
+test('.set() - with Date', t => {
+	t.context.config.set('dateobj', new Date());
+	const myDate = t.context.config.get('dateobj');
+	t.true(myDate instanceof Date);
+});
+
+test('.set() - with Date in a nested object', t => {
+	t.context.config.set('dateobj', {str: 'this is a string', date: new Date()});
+	const myDate = t.context.config.get('dateobj').date;
+	t.true(myDate instanceof Date);
+});
```

Comment:
I've added more tests, but I'm not sure how to go about circular objects, becuase `Conf.serialize`, which is `JSON.stringify` by default, throws an exception is there's a circulation. So all I did for now is to detect if there's a circulation and stop handling that branch of the object, because it would cause an infinite loop, and cause the application to crash, but when `Conf` gets to serializing the object with `JSON` it still throws an error. I've also tested this behavior with `string` so it's not a problem with how the new feature handles extra types.